### ISSUE TITLE
CMS URL

### ISSUE THREAD

**Issue by trangng** [sphereio/sphere-sunrise#197]
Hi @PeterRoman,

we decided on the following URLs for

- CMS sunrise.commercetools.de/:ang-optional-country/:slug-for-lang-and-country

- and category sunrise.commercetools.io/language/category-slug/

The problem here, we can not differentiate between category and CMS and can not differentiate which controller should be used.

One solution would be to use a prefix e.g. "collision". Would that influence SEO or is there any approach you would suggest from the SEO side?

**Comment by trangng** [comment]
@PeterRoman any news on this?

**Comment by trangng** [comment]
@PeterRoman could you please give feedback to this issue. thx

**Comment by schleichardt** [comment]
Sorry, this is on me, Peter had technical questions, but due to deadlines and office absence we could not meet yet. @PeterRoman let's discuss this on Tuesday next week.

**Comment by trangng** [comment]
ok thanks for the update @schleichardt

**Comment by PeterRoman** [comment]
Hey Trang.

- it should be commercetools.com instead of .de tld?!
- according to Michael both urls (cms and category) are clearly distinguishable: trailing slash for category slug. 
- what about .html at the end of the cms url?
- To answer your question: maybe there would be a SEO effect if we add subdirectories. We want to avoid any prefixes not only for SEO but for the best user experience. For example:
In the grid shops the amount of subdirectories and therefore redirects is not beneficial for SEO: coeur.de >> https://www.coeur.de/cdl/de/cms/welcome

**Comment by lauraluiz** [comment]
For me this is okay:
```
/:languageTag/:cmsSlug.html
/:languageTag/:categorySlug/:productSlug-:sku.html
/:languageTag/:categorySlug/
```
Do we agree with this?

**Comment by schleichardt** [comment]
Can skus contain a `-`?

**Comment by lauraluiz** [comment]
No. I simplify it here for better understanding, but the Play route would be as we currently have it:
```
/:languageTag/:productSlug-$sku<[^-]+>.html
```

**Comment by jayS-de** [comment]
As a product can have more than one category, which category should be used for a product url?

/:languageTag/:categorySlug/:productSlug-:sku.html

Also SKUs may contain a "-". Maybe not for our sunrise data, but for other projects.

**Comment by lauraluiz** [comment]
About the SKUs, yes, certainly we should allow them, but for that the only solution is using one more folder level:
```
/:languageTag/:categorySlug/:productSlug/:sku.html
```
How is that concerning SEO @PeterRoman ?

**Comment by jayS-de** [comment]
Routes i'm declared so far:

/:locale/ -  catalog.controller:home
/:locale/search/ - catalog.controller:search
/:locale/cart/add - cart.controller:add
/:locale/cart/delete - cart.controller:deleteLineItem
/:locale/cart/change - cart.controller:changeLineItem
/:locale/cart - cart.controller:index
/:locale/category - catalog.controller:search
/:locale/:slug.html - catalog.controller:detail
/:locale/:slug/:sku.html - catalog.controller:detail

**Comment by lauraluiz** [comment]
Yes, actually I didn't see that we didn't have category with the product URL before. With the SKU it fits well for both technical and SEO reasons. Or so I hope :)
```
/:languageTag/:cmsSlug.html
/:languageTag/:productSlug/:sku.html
/:languageTag/:categorySlug/
```

**Comment by trangng** [comment]
ok agreed on this structure above. 

@PeterRoman we could use sunrise.commercetools.com, that should not be a problem. Actually we could have both and redirect de to com. I'll check with ops.
